### ISSUE TITLE
nbgrader: add another option for where to run nbgrader

### ISSUE THREAD

**Issue by williamstein** [sagemathinc/cocalc#4877]
Right now you have two options about where to run nbgrader, determined by this checkbox in course config:

![image](https://user-images.githubusercontent.com/1276278/93890068-cba76900-fc9e-11ea-95dc-7035774a03b2.png)

It would make a lot of sense to have a third option, where you specify a specific project (of your choosing) in which to run nbgrader.  Why?

1. You can upgrade this project precisely how you want.
2. You can put exactly the files in this project you want.
3. You don't expose any info in your instructor project to potentially malicious code.
4. You avoid wastefully starting tons of student projects to accomplish grading of their work.

It seems like an ideal solution.  Even better would be a "pool" of your own projects, and you can run nbgrader in parallel across that pool.  But maybe that is for another ticket.
